test/Microsoft.AspNetCore.Razor.Language.Test/DefaultRazorTagHelperBinderPhaseTest.cs

```diff
@@ -117,11 +111,8 @@ public void Execute_CanHandleSingleLengthTagHelperPrefix()
             };
             var expectedDiagnostics = new[]
             {
-                RazorDiagnostic.Create(
-                    new RazorError(
-                        LegacyResources.ParseError_Unterminated_String_Literal,
-                        new SourceLocation(17 + Environment.NewLine.Length, 1, 17),
-                        length: 1)),
+                RazorDiagnosticFactory.CreateParsing_UnterminatedStringLiteral(
```

Comment:
**non-blocking:**

@NTaylorMullen and @ajaybhargavb do you guys like this pattern of `Assert.Equals` + expected?

Another approach would be to write an assert method for it that takes the descriptor and span.

Reply:
I'm personally a fan of verifying the entire error for 1 reason, if the diagnostic takes in parameters you end up needing to substitute those into the descriptor anyways to have accurate verification.

Reply:
In that case you just verify the parameters as well.

Reply:
This works for tests where we just verify the errors directly but in a lot of other cases where we build the expected root of a big syntax tree with a bunch of expected errors and compare it with the actual root like [here](https://github.com/aspnet/Razor/blob/dev/test/Microsoft.AspNetCore.Razor.Language.Test/Legacy/CSharpBlockTest.cs#L502), it might complicate our test infrastructure more than we want.
Plus, we don't have access to the individual parameters once the `RazorDiagnostic` has been created. So I'm not sure how they can be compared.